### PR TITLE
Upgrade MaterialDrawer

### LINKED ISSUE #1335: MaterialDrawer upgrade
`com.mikepenz.materialdrawer:library:3.0.6@aar` is available and with it, apparently `com.mikepenz.iconics:library:0.9.1@aar` won't be needed anymore.

Changes in code are minimal and I'd happily upgrade it and possibly other libraries to the latest versions, unless there are policies about library upgrades in place.

Furthermore, `MaterialDrawerTheme.Light.ActionBar` can be used as parent for `KeychainTheme.Base`, as it itself extends `KeychainTheme.Base`'s current parent `Theme.AppCompat.Light`. This would allow tweaking of the color scheme of the Drawer by overriding `material_drawer_*` colors.

### PR DESCRIPTION
**Pull request by thialfihar**
Since MaterialDrawer also grabs `support-v4` and `appcompat-v7` the hashes differ. Everything seems to work fine, but since I don't know how strict you are about the hash verification, you best decide whether that's an issue. :)

Fixes #1335.

### CODE REVIEW

File: OpenKeychain/build.gradle [hunk 1]
```diff
@@ -43,8 +43,9 @@ dependencies {
     compile "com.splitwise:tokenautocomplete:1.3.3@aar"
     compile 'se.emilsjolander:stickylistheaders:2.6.0'
     compile 'org.sufficientlysecure:html-textview:1.1'
-    compile 'com.mikepenz.materialdrawer:library:2.8.2@aar'
-    compile 'com.mikepenz.iconics:library:0.9.1@aar'
+    compile('com.mikepenz:materialdrawer:3.0.7@aar') {
+        transitive = true
```
Comment (dschuermann) [inline OpenKeychain/build.gradle]: transitive dependencies here makes it difficult to verify sha256 checksums later. materialdrawer depends on appcompat-v7:22.2.0 and com.mikepenz:iconics:1.0.2 (see https://repo1.maven.org/maven2/com/mikepenz/materialdrawer/3.0.7/materialdrawer-3.0.7.pom )
Please include iconics and update appcompat-v7

**Comment by thialfihar** [comment]
Maybe let this ripen for a while, actually. It seems to change the text color of the Settings title for some reason. I'll investigate further sometime soon.

File: OpenKeychain/build.gradle (hunk 1 above)
Comment (thialfihar) [inline OpenKeychain/build.gradle]: Ok!

File: OpenKeychain/src/main/res/values/themes.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
     <style name="KeychainTheme" parent="KeychainTheme.Base" />
 
-    <style name="KeychainTheme.Base" parent="Theme.AppCompat.Light">
+    <style name="KeychainTheme.Base" parent="MaterialDrawerTheme.Light.ActionBar">
```
Comment (dschuermann) [inline OpenKeychain/src/main/res/values/themes.xml]: Maybe then do this pull request without this change for now?

Comment (thialfihar) [inline OpenKeychain/src/main/res/values/themes.xml]: No, no. I already removed that, thinking it was that. But even with `Theme.AppCompat.Light` it changes the color. Very strange. I'll look into it.

**Comment by Valodim** [comment]
closing for now to keep issue tracker clean, please reopen when relevant

**Pull request closed without merging**